ledger/src/blockstore.rs

```diff
@@ -1294,6 +1294,10 @@ impl Blockstore {
             false
         };
 
+        if shred.data_header.size == 0 {
```

Comment:
@carllin think we need cutoff for this? I don't think we create a len=0 shred in the normal code paths.

Reply:
Yeah, you're thinking it might actually be useful to store the zero length shred for duplicate detection? 

If we store it most likely I think replay will call `get_slot_entries()`, the best case scenario is if the validator then can't recreate the entries and mark the slot as dead. 

This brings up an interesting point though. I wonder if it's possible to use zero-length shreds as filler to recreate the same entries, but with a different set of shreds, i.e. instead of splitting an entry amount `N` shreds, you spread it out between `N + Z` shreds where `Z` is some number of empty shreds (could also do this without zero-length shreds I think).

I'm guessing an attacker could do this with just the last couple of shreds to optimize their chances that some validators getting all of one version or another (you don't want a mix of versions otherwise they won't deserialize properly).

Replay might actually play that alternate version of the shreds without a hitch. As long as you end up with the right state though, it should be fine? It might make repair a little wonky if some people are serving one version of a slot, and others serving another version though.

Reply:
I am not sure this change was backward compatible.
broadcast has to create zero length shreds when the previous slot was interrupted:
https://github.com/solana-labs/solana/blob/ca7b36ad8/core/src/broadcast_stage/standard_broadcast_run.rs#L79

`shred data == None` will result in zero length data shred.

Reply:
Per some back and forth on Discord, we believe this is a non-issue as the the headers will take up some space. Behzad suggested adding a unit test to protect the behavior we rely on; that test is added here:
https://github.com/solana-labs/solana/pull/16955

Reply:
awesome, thanks for reasoning through it guys 😃